modules/images/regenerate-existing-images/background-process/class-perflab-background-job.php

```diff
@@ -0,0 +1,454 @@
+<?php
+/**
+ * Background job class.
+ *
+ * @since n.e.x.t
+ * @package performance-lab
+ */
+
+/**
+ * Class Perflab_Background_Job.
+ *
+ * Manage and run the background jobs.
+ *
+ * @since n.e.x.t
+ */
+class Perflab_Background_Job {
+	/**
+	 * Meta key for storing job name/action.
+	 *
+	 * @const string
+	 * @since n.e.x.t
+	 */
+	const JOB_NAME_META_KEY = 'perflab_job_name';
+
+	/**
+	 * Meta key for storing job data.
+	 *
+	 * @const string
+	 * @since n.e.x.t
+	 */
+	const JOB_DATA_META_KEY = 'perflab_job_data';
+
+	/**
+	 * Meta key for storing job data.
+	 *
+	 * @const string
+	 * @since n.e.x.t
+	 */
+	const JOB_RETRY_META_KEY = 'perflab_job_retries';
+
+	/**
+	 * Meta key for storing job lock.
+	 *
+	 * @const string
+	 * @since n.e.x.t
+	 */
+	const JOB_LOCK_META_KEY = 'perflab_job_lock';
+
+	/**
+	 * Meta key for storing job errors.
+	 *
+	 * @const string
+	 * @since n.e.x.t
+	 */
+	const JOB_ERRORS_META_KEY = 'perflab_job_errors';
+
+	/**
+	 * Job status meta key.
+	 *
+	 * @const string
+	 * @since n.e.x.t
+	 */
+	const JOB_STATUS_META_KEY = 'perflab_job_status';
+
+	/**
+	 * Job status for queued jobs.
+	 *
+	 * @const string
+	 * @since n.e.x.t
+	 */
+	const JOB_STATUS_QUEUED = 'perflab_job_queued';
+
+	/**
+	 * Job status for running jobs.
+	 *
+	 * @const string
+	 * @since n.e.x.t
+	 */
+	const JOB_STATUS_RUNNING = 'perflab_job_running';
+
+	/**
+	 * Job status for completed jobs.
+	 *
+	 * @const string
+	 * @since n.e.x.t
+	 */
+	const JOB_STATUS_COMPLETE = 'perflab_job_complete';
+
+	/**
+	 * Job status for failed jobs.
+	 *
+	 * @const string
+	 * @since n.e.x.t
+	 */
+	const JOB_STATUS_FAILED = 'perflab_job_failed';
+
+	/**
+	 * Job ID.
+	 *
+	 * @var int
+	 */
+	public $job_id;
+
+	/**
+	 * Job name.
+	 *
+	 * @var string
+	 */
+	public $name;
+
+	/**
+	 * Job data.
+	 *
+	 * @var array
+	 */
+	public $data;
+
+	/**
+	 * Perflab_Background_Job constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int $job_id job ID.
+	 */
+	public function __construct( $job_id = 0 ) {
+		$this->job_id = absint( $job_id );
+
+		// @todo Replace taxonomy name with constant.
+		if ( term_exists( $this->job_id, 'background_job' ) ) {
```

Comment:
We already have `exists` method so use it here.
```suggestion
		if ( $this->exists() ) {
```